### DISCUSSION TITLE
[RESOLVED] How should one use SinkModelCsv?

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [github/codeql#8456] [General]
I have defined a class that inherits SinkModelCsv:

```
private class CommandSinkModelCsv extends SinkModelCsv {
  override predicate row(string row) {
    row =
    [
      "java.lang;Runtime;true;exec;;;Argument[0];cmdi",
      "java.lang;ProcessBuilder;true;ProcessBuilder;;;Argument[0];cmdi",
    ]
  }
}
```

Problems:

1. Can I match ProcessBuilder constructor with SinkModelCsv?
2. How should I use it in the `TaintTracking::Configuration` class, e.g `override predicate isSink(DataFlow::Node sink)`

**Comment by hvitved** [comment]
`SinkModelCsv` (and everything else in `ExternalFlow.qll`) is meant for internal use only (it will be made public, once we have a more stable API), so I strongly suggest not using it just yet.

**Reply by CaledoniaProject** [reply]
Understood.
